File: pyclient/integtest.py
#!/usr/bin/python

from lockd import LockdClient

lockd_client = LockdClient()

# Initial state
assert not lockd_client.is_locked("foo")
assert not lockd_client.is_locked("bar")

# Lock
assert lockd_client.lock("foo")
assert lockd_client.is_locked("foo")

# Dup lock should fail
assert not lockd_client.lock("foo")
assert lockd_client.is_locked("foo")

# Lock another entry should work
assert lockd_client.lock("bar")
assert lockd_client.is_locked("bar")

# Unlock entries
assert lockd_client.unlock("foo")
assert lockd_client.unlock("bar")
assert not lockd_client.is_locked("foo")
assert not lockd_client.is_locked("bar")

# Dup unlock should fail
assert not lockd_client.unlock("bar")
assert not lockd_client.is_locked("bar")

File: integtest/integtest.go
package main

import (
	"log"

	"github.com/divtxt/lockd/lockd_client"
)

func main() {
	lc := lockd_client.NewLockdClient()

	// Initial state
	assert(lc.IsLocked, "foo", false)
	assert(lc.IsLocked, "bar", false)

	// Lock
	assert(lc.Lock, "foo", true)
	assert(lc.IsLocked, "foo", true)

	// Dup lock should fail
	assert(lc.Lock, "foo", false)
	assert(lc.IsLocked, "foo", true)

	// Lock another entry should work
	assert(lc.Lock, "bar", true)
	assert(lc.IsLocked, "bar", true)

	// Unlock entries
	assert(lc.Unlock, "bar", true)
	assert(lc.Unlock, "foo", true)
	assert(lc.IsLocked, "foo", false)
	assert(lc.IsLocked, "bar", false)

	// Dup unlock should fail
	assert(lc.Unlock, "bar", false)
	assert(lc.IsLocked, "bar", false)

}

func assert(f func(string) (bool, error), name string, expected bool) {
	actual, err := f(name)
	if err != nil {
		log.Panic(err)
	}
	if actual != expected {
		log.Panicf("Actual: %v != Expected: %v", actual, expected)
	}
}
